### DISCUSSION TITLE
Avoid loss of cached tiles when changing token

### DISCUSSION THREAD

**Discussion by bmason8** [JaffaKetchup/flutter_map_tile_caching#127] [Q&A] [answered]
I needed to change my mapbox token and noticed that when I did, all my cached tiles were still stored but no longer accessible because (I think) the filename includes the token in it. Switching back to my old token made those cached tiles accessible again but this causes an obvious issue.

Is there a tool for updating the filenames to account for a token change or am I forced to delete the stored tiles and alert the user they need to download them again? I read through the change log of the versions above what I'm using (6.2.0) but don't see anything that suggests this has been accounted for).

Any help would be appreciated!

Using version:  _flutter_map_tile_caching: ^6.2.0_

environment:
  _sdk: '>=2.17.0 <3.0.0'
  flutter: ">=3.0.0"_

flutter sdk: _3.3.7_

**Comment by JaffaKetchup** [comment, marked as answer]
Hey again @bmason8,

[Since v7.1.0](https://github.com/JaffaKetchup/flutter_map_tile_caching/blob/main/CHANGELOG.md#710---20230214) & in the 'v6-backporting' branch, the "URL query params obscurer" feature is available, which sounds like what you might need. See https://fmtc.jaffaketchup.dev/usage/integration#obscuring-query-parameters for documentation.

Let me know if you need anything else!

**Reply by bmason8** [reply]
Awesome, thanks @JaffaKetchup, that looks like what I need!
